app/controllers/projects_controller.rb

```diff
@@ -59,7 +59,7 @@ def create_project
 
   def update_project
     @update_project ||=
-      UpdateProject.call(project_params: project_params, current_user: current_user, project_id: @project.id)
+      UpdateProject.call(project_params: project_params, current_user: current_user, project_obj: @project)
```

Comment:
is there a reason for calling this parameter `project_obj` and not just `project`?
```suggestion
      UpdateProject.call(project_params: project_params, current_user: current_user, project: @project)
```

Reply:
When I use project (not the project_obj), I get this error
![image](https://user-images.githubusercontent.com/70964774/143605426-496a55b5-d551-412d-8ce3-3075d19d7e80.png)

Reply:
I think this is because the project variable already exists

Reply:
As I understand after you fixes, now you will not get this error. Try to use project instead of project_obj.

Reply:
This is because you are calling the method `project` inside itself unconditionally.